### PR TITLE
Add mainnet to wCCD example

### PR DESCRIPTION
**Pull request by DOBEN**
## Purpose

Add support for wCCD frontend for mainnet. wCCD is now deployed on mainnet and we would like to host this frontend for it as well.

## Changes

The front end can either be built for mainnet or testnet as described in READ.me. Equally, the `dockerFile` can be built for mainnet or testnet.

- If the user uses the browser wallet, an error message will appear now if the browser wallet is set to the wrong network.
- If the user uses the mobile wallet (Android), an error message will appear now if the wrong (for the wrong network) mobile wallet app is used.

### CODE REVIEW

File: examples/wCCD/src/constants_mainnet_network.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// The MAINNEt_GENESIS_BLOCK_HASH is used to check that the user has its browser wallet connected to testnet and not to mainnet.
+import { Network } from '@concordium/react-components';
+
+export const MAINET_GENESIS_BLOCK_HASH = '9dd9ca4d19e9393877d2c44b70f89acbfc0883c2243e5eeaecc0d1cd0503f478';
```
Comment (abizjak) [inline examples/wCCD/src/constants_mainnet_network.ts:4]: It's usually spelled "MAINNET".

**Review by abizjak — COMMENTED** [review]
I think the Jenkinsfile also must be updated to take into account  the new build parameter.

File: examples/wCCD/src/constants_network.ts
[the remark this answers is not in the document]
Comment (DOBEN) [reply examples/wCCD/src/constants_network.ts:1]: I tried that but run into the issue that building of the frontend works but once run in the browser wallet I get this error `process` not defined. Is there a specific environmental variable framework that needs to be used to get this working?

[the remark this answers is not in the document]
Comment (shjortConcordium) [reply examples/wCCD/src/constants_network.ts:1]: The `process.env.ENV_VAR_NAME` only "exists" during "compilation", so you have to replace it in the build script.
Simplest way is in esbuild.config.ts to add:
```
const config: BuildOptions = {
    ...
    define: {
        ...
        'process.env.network': process.env.network
    },
}
```
So access the value during in the app.

File: examples/wCCD/src/wCCD.tsx [hunk 2]
```diff
@@ -133,10 +133,23 @@ export default function wCCD(props: WalletConnectionProps) {
     const [admin, setAdmin] = useState<string>();
     const [adminError, setAdminError] = useState('');
 
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const testnet = 'testnet';
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const mainnet = 'mainnet';
```
Comment (shjortConcordium) [inline examples/wCCD/src/wCCD.tsx:139]:
```suggestion
```
Or why are you keeping them? :thinking:

**Review by shjortConcordium — APPROVED** [review]

**Review by abizjak — APPROVED** [review]

File: examples/wCCD/src/wCCD.tsx (hunk 2 above)
Comment (DOBEN) [reply examples/wCCD/src/wCCD.tsx]: `process.env.NETWORK` is replaced by e.g. testnet (not by the string `testnet`) and without defining testnet=`testnet`, I got an error that testnet variable is not defined.

**Review by shjortConcordium — APPROVED** [review]

**Pull request merged**